Make repeatable field delete confirmation optional

barryceelen opened the pull request:
Currently a javascript confirm modal dialog is shown when before removing a repeatable field.
- Can become tedious when removing a couple of items at a time.
- In Chrome the user can check the ‘Prevent this page from creating additional dialogs’ option in the confirmation dialog. With the result of not being able to delete any more items (Until browser restart/window open/close/history reset?).

![pasted_image_11_09_16_23_19](https://cloud.githubusercontent.com/assets/4179791/18421065/0f85585e-7881-11e6-8c4f-44a6d9613ab1.jpg)

This pull request adds a `confirm_delete` option which defaults to true.

mikeselander commented on class.cmb-meta-box.php:
```diff
@@ -295,6 +295,9 @@ static function layout_fields( array $fields ) { ?>
 					$attrs[] = sprintf( 'data-rep-max="%s"', intval( $field->args['repeatable_max'] ) );
 
+				// Ask for confirmation before removing field.
+				if ( ! empty( $field->args['confirm_delete_field'] ) )
```
Could you please wrap this in brackets?

mikeselander replied:
If the user sets this to false, the data attribute won't fill in - Since we're defaulting I'd rather see a check that doesn't also apply to `false` values.

barryceelen commented:
@mikeselander Hope this addresses the issues.

Note: I’m now checking for `false !== field.data('confirm-delete’)` in stead of just `true` in case the data attribute is not present at all for some reason.

mikeselander commented:
Great, thanks @barryceelen - I'll take a look sometime this week 👍

mikeselander commented:
Looking good - merging 👍
